katana/native/cases/static/cases/js/casesApp.js

```diff
@@ -0,0 +1,1040 @@
+			
+			ta += 1
+			a_items.push('</div>');
+			
+		}
+		popup.find("#arguments-textarea").html( a_items.join("\n"));	
+		if (!jQuery.isArray(caseApp.jsonCaseSteps['step'])) {
+			caseApp.jsonCaseSteps['step'] = [caseApp.jsonCaseSteps['step']];
+			}
+		caseApp.fillCaseStepDefaultGoto(popup);
+			popup.find('#SteponError-at-action').on('change', function(){ 
+				var popup = $(this).closest('.popup');
+				fillCaseStepDefaultGoto(popup);
+		});
+
+	},
+
+
+// Fill in default GoTo steps for a popup window for a case step. 
+	fillCaseStepDefaultGoto : function(popupx) {
+	var popup = $(this).closest('.popup');
+	var gotoStep =popup.find('#SteponError-at-action').val();
+	console.log("Step ", gotoStep, popup.find('#SteponError-at-action'));
+	var defgoto = popup.find('#SteponError-at-value'); 
+	defgoto.hide();
+
+		// if (gotoStep.trim() == 'goto'()) { 
+		// 	defgoto.show();
+		// } else {
+		// 	defgoto.hide();
+			
+		// }
+
+
+		//var sid = popup.find('#CaseRowToEdit').val();
+		defgoto.empty(); 
+		var xdata = caseApp.jsonCaseObject["Steps"]['step']; // ['Testcase']; 
+		console.log("Step....",Object.keys(xdata).length, xdata);
+		for (var s=0; s<Object.keys(xdata).length; s++ ) {
+			defgoto.append($('<option>',{ value: s,  text: s+1}));
+		}
+	},
+
+
+// Show a popup dialog with items from a case.
+	setupPopupDialog: function (sid,xdata,popup) {
+		console.log(popup);  // Merely returns the div tag
+		var dd_driver = popup.find('#StepDriver');
+		oneCaseStep = xdata[sid]
+		var driver = oneCaseStep[ "@Driver"]  // 
+		var keyword  = oneCaseStep[ "@Keyword"]   // 
+		var a_items = []; 
+		console.log("---- oneCase ---- ", oneCaseStep["@Driver"], oneCaseStep[ "@Keyword"] , oneCaseStep);
+		popup.attr("caseStep", oneCaseStep);
+		popup.attr("sid", sid);
+		jQuery.getJSON("./cases/getListOfActions").done(function(data) {
+				a_items = data['actions'];
+				dd_driver.empty();  // Empty all the options....
+				for (var x =0; x < a_items.length; x++) {
+						dd_driver.append($('<option>',{ value: a_items[x],  text: a_items[x]}));
+					}
+				//console.log(dd_driver.html());
+				popup.find('#StepDriver').val(oneCaseStep["@Driver"]);
+				popup.find("#StepKeyword").val(oneCaseStep["@Keyword"]);
+				// Now set up the keywords
+				var driver = oneCaseStep[ "@Driver"]  ;
+				var keyword  = oneCaseStep[ "@Keyword"];
+				console.log("Collecting ...", driver, keyword);
+
+				jQuery.getJSON("./cases/getListOfKeywords/?driver="+driver).done(function(data) {
+ 						popup.find("#StepKeyword").empty();
+ 						a_items = data['keywords'];
+ 						console.log(a_items);
+ 						for (let x of a_items) {
+ 							popup.find("#StepKeyword").append($('<option>',{ value: x,  text: x }));
+ 						}
+ 					popup.find('#StepKeyword').val(oneCaseStep["@Keyword"]);
+				
+					jQuery.getJSON("./cases/getListOfComments/?driver="+driver+"&keyword="+keyword).done(function(data) {
+	 					a_items = data['fields'];
```

Comment:
Get keywords and associated data like comments, signature etc at the same time instead of a separate call.

Reply:
Done